### PR TITLE
[ALLUXIO-2115] Add config for request signature used by s3 and upgrade jets3t

### PR DESCRIPTION
**Pull request by LeightonWong**
https://alluxio.atlassian.net/browse/ALLUXIO-2115

* Upgrade aws-java-sdk to 1.11.20
* Upgrade jets3t to 0.9.4 to support AWS4-HMAC-SHA256
* Add config _alluxio.underfs.s3.request.signature.version_ to tell jets3t to use signature v4 for request authentication.

Because [all regions support Signature Version 4](http://docs.aws.amazon.com/general/latest/gr/sigv4_support.html), so we also add config to alluxio-default.properties

### CODE REVIEW

**Comment by LeightonWong** [comment]
alluxio-bot, check this please

**Comment by LeightonWong** [comment]
alluxio-bot, check this please

File: pom.xml [hunk 1]
```diff
@@ -139,7 +139,7 @@
 
   <properties>
     <apache.curator.version>2.1.0-incubating</apache.curator.version>
-    <aws.amazonaws.version>1.11.7</aws.amazonaws.version>
+    <aws.amazonaws.version>1.11.20</aws.amazonaws.version>
```
Comment (calvinjia) [inline pom.xml]: Is this version change required to support V4?

File: underfs/pom.xml [hunk 2]
```diff
@@ -39,7 +39,7 @@
       <dependency>
         <groupId>net.java.dev.jets3t</groupId>
         <artifactId>jets3t</artifactId>
-        <version>0.8.1</version>
+        <version>0.9.4</version>
```
Comment (calvinjia) [inline underfs/pom.xml]: v0.9.x jets3t has performance issues when doing seeks. See: https://bitbucket.org/jmurty/jets3t/commits/b41fbb3aa606
Is it possible to use the S3A client with V4?

**Comment by calvinjia** [comment]
@LeightonWong Thanks for the patch, I left a few comments.

File: underfs/pom.xml (hunk 2 above)
Comment (LeightonWong) [inline underfs/pom.xml]: Yes, I made a little modify to s3a module, to accept user define entry point like “s3.cn-north-1.amazonaws.com.cn” and it works perfect for signature v4 request authentication. Should I make another pr only include those s3a project changes?

File: pom.xml (hunk 1 above)
Comment (LeightonWong) [inline pom.xml]: It is not required.

File: underfs/pom.xml (hunk 2 above)
Comment (calvinjia) [inline underfs/pom.xml]: That sounds good. Are those changes in this PR (I only see changes to S3UnderFileSystem and not S3AUnderFileSystem)?

Comment (LeightonWong) [inline underfs/pom.xml]: Just commit.

**Comment by calvinjia** [comment]
@LeightonWong The S3A change looks good. Could you revert the other changes (s3n changes, version increase), and add documentation for the new configuration value? See [this commit](https://github.com/Alluxio/alluxio/pull/3656/commits/ed9fb56606c9182d175dd68214092eb3a7cebffd) as an example.

**Comment by calvinjia** [comment]
@LeightonWong Also, I don't think your github account is probably linked yet. For us to properly credit your contributions, could you please set up your Github account by following the Github instructions [here](https://help.github.com/articles/setting-your-email-in-git/)?

**Comment by LeightonWong** [comment]
@calvinjia Ok, s3n changes reverted, document added, account linked.

**Comment by calvinjia** [comment]
LGTM, thanks for the iterations.

Could you sign the CLA (see the bot message)? I think this is good to merge.

**Comment by LeightonWong** [comment]
@calvinjia OK, already signed.

**Comment by calvinjia** [comment]
alluxio-bot, check this please.

**Comment by calvinjia** [comment]
@LeightonWong 
Could you double check you have signed the CLA? After you sign it, the bot should say a message like `You did it! Thanks for signing the CLA` and the `cla-missing` label will be removed from this PR. Thanks!

**Comment by LeightonWong** [comment]
@calvinjia Sorry， I checked status and resigned, now cla-missing removed.

**Comment by calvinjia** [comment]
LGTM
